samza-core/src/main/java/org/apache/samza/container/grouper/task/SimpleGroupByContainerCount.java

```diff
@@ -0,0 +1,102 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.samza.container.grouper.task;
+
+import java.util.Arrays;
+import org.apache.samza.container.TaskName;
+import org.apache.samza.job.model.ContainerModel;
+import org.apache.samza.job.model.TaskModel;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+
+/**
+ * Simple grouper.
+ * It exposes two group methods - one that assumes sequential container numbers and one that gets a set of container
+ * IDs as an argument. Please note - this first implementation ignores locality information.
+ */
+public class SimpleGroupByContainerCount implements TaskNameGrouper {
+  private final int startContainerCount;
+  public SimpleGroupByContainerCount() {
+    this.startContainerCount = 1;
+  }
+
+  public SimpleGroupByContainerCount(int containerCount) {
+    if (containerCount <= 0) throw new IllegalArgumentException("Must have at least one container");
+    this.startContainerCount = containerCount;
+  }
+
+  @Override
+  public Set<ContainerModel> group(Set<TaskModel> tasks) {
+    if(tasks.isEmpty())
+      throw new IllegalArgumentException("cannot group an empty set");
+
+    if(startContainerCount > tasks.size())
+      throw new IllegalArgumentException("number of containers="  + startContainerCount + " is bigger than number of tasks=" + tasks.size());
+
+    List<Integer> containerIds = new ArrayList<>(startContainerCount);
+    for (int i = 0; i < startContainerCount; i++) {
+      containerIds.add(i);
+    }
+    return group(tasks, containerIds);
+  }
+
+  public Set<ContainerModel> group(Set<TaskModel> tasks, List<Integer> containersIds) {
+    if(tasks.isEmpty())
+      throw new IllegalArgumentException("cannot group an empty set. containersIds=" + Arrays
+          .toString(containersIds.toArray()));
+
+    if(containersIds.size() > tasks.size())
+      throw new IllegalArgumentException("number of containers "  + containersIds.size() + " is bigger than number of tasks " + tasks.size());
+
+    if (containersIds == null)
+      return this.group(tasks);
+
+    int containerCount = containersIds.size();
+
+    // Sort tasks by taskName.
+    List<TaskModel> sortedTasks = new ArrayList<>(tasks);
+    Collections.sort(sortedTasks);
+
+    // Map every task to a container in round-robin fashion.
+    Map<TaskName, TaskModel>[] taskGroups = new Map[containerCount];
+    for (int i = 0; i < containerCount; i++) {
+      taskGroups[i] = new HashMap<>();
+    }
+    for (int i = 0; i < sortedTasks.size(); i++) {
+      TaskModel tm = sortedTasks.get(i);
+      taskGroups[i % containerCount].put(tm.getTaskName(), tm);
+    }
+
+    // Convert to a Set of ContainerModel
+    Set<ContainerModel> containerModels = new HashSet<>();
```

Comment:
What is the reason of using a set instead of an array for containerModels? If we are using the array, we just keep the taskGroups[i] for each element of the array, and it seems better to me for the quick access on index.